blockstore/context.go

```diff
@@ -0,0 +1,21 @@
+package blockstore
+
+import (
+	"context"
+)
+
+type hotViewKey struct{}
+
+var hotView = hotViewKey{}
+
+// WithHotView constructs a new context with an option that provides a hint to the blockstore
+// (e.g. the splitstore) that the object (and its ipld references) should be kept hot.
+func WithHotView(ctx context.Context) context.Context {
+	return context.WithValue(ctx, hotView, struct{}{})
+}
+
+// GetHotView returns true if the hot view option is set in the context
+func GetHotView(ctx context.Context) bool {
```

Comment:
Nit: Maybe `IsHotView` as we're really just checking if we're in one.

Reply:
sure.

Reply:
renamed to `IsHotView`.